src/components/Todos/CreateTodoForm.js

```diff
@@ -0,0 +1,207 @@
+        value: "",
+        isvalid: null,
+    });
+
+    {
+        /**define useReducer for description input */
+    }
+    const [descState, dispatchDesc] = useReducer(descReducer, {
+        value: "",
+        isvalid: null,
+    });
+
+    {
+        /**make variable from State as arrugument of useEffect */
+    }
+    const { isvalid: titleIsValid } = titleState;
+    const { isvalid: descIsValid } = descState;
+
+    {
+        /**check form validation state's boolean in order to enable submit button to push or disable
+         * whenever each input state changes*/
+    }
+    useEffect(() => {
+        const identifier = setTimeout(() => {
+            setFormIsvalid(titleIsValid && descIsValid);
+        }, 500);
+        //cleanup
+        return () => {
+            clearTimeout(identifier);
+        };
+    }, [titleIsValid, descIsValid]);
+
+    {
+        /**trigger fn to check titile input state on value change */
+    }
+    const titleChangeHandler = (event) => {
+        dispatchTitle({ type: "USER_INPUT", value: event.target.value });
+    };
+
+    {
+        /**trigger fn to check desc input state on value change */
+    }
+    const descChangeHandler = (event) => {
+        dispatchDesc({ type: "USER_INPUT", value: event.target.value });
+    };
+
+    {
+        /**trigger fn to check titile input state on focus blurred from input */
+    }
+    const validateTitleHandler = () => {
+        dispatchTitle({ type: "INPUT_BLUR" });
+    };
+
+    {
+        /**trigger fn to check desc input state on focus blurred from input */
+    }
+    const validateDescHandler = () => {
+        dispatchDesc({ type: "INPUT_BLUR" });
+    };
+
+    {
+        /**on submit, post new created todo to API server */
+    }
+    const submitHandler = (e) => {
+        e.preventDefault();
+
+        setIsLoading(true);
+        {
+            /**data body */
+        }
+        const newTodo = {
+            title: titleState.value,
+            description: descState.value,
+        };
+
+        const createAPI = "http://localhost:5151/todos";
+        axios
+            .post(createAPI, newTodo, {
+                headers: {
+                    "Content-Type": "application/json",
+                    Authorization: token,
+                },
+            })
+            .then((response) => {
+                {
+                    /**if process end successfully, change loading state & redirect to todoList page */
+                }
+                if (response.status === 200) {
+                    history.push("/todos");
+                }
+                {
+                    /**if process failed, show message to tell that */
+                }
+                setMessage("Create Todo failed.");
+            })
+            .catch((error) => {
+                {
+                    /**store some error in this process */
+                }
+                setMessage(error.message);
```

Comment:
Just notice: Imagine that token is expired, server will return status code 401 here (Unauthorized),  and you should show more detail message like `you have no permission to perform this action` or redirect user to login page.